### ISSUE TITLE
Setting `reply_broadcast` gives a type error when it's a dynamic value

### ISSUE THREAD

**Issue by Amethystix** [slackapi/node-slack-sdk#1859]
#### Packages:

Select all that apply:

- [x] `@slack/web-api`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [ ] `@slack/oauth`
- [ ] `@slack/socket-mode`
- [ ] `@slack/types`
- [ ] I don't know

#### The Slack SDK version
v7.3.1

#### Node.js runtime version
18.18.2

#### Typescript version
5.5.4

#### OS info
```
ProductName:		macOS
ProductVersion:		13.3.1
ProductVersionExtra:	(a)
BuildVersion:		22E772610a
Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:28 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6000
```

#### Steps to reproduce:
Dynamically set `reply_broadcast` as we do in this wrapper function:
```
  async reply(
    channelId: string,
    message: Block[] | string,
    threadId: string,
    broadcast = false,
  ) {

    if (typeof message === "string") {
      return await this.web.chat.postMessage({
        channel: channelId,
        thread_ts: threadId,
        text: message,
        reply_broadcast: broadcast,
      });
    } else {
      return await this.web.chat.postMessage({
        channel: channelId,
        thread_ts: threadId,
        reply_broadcast: broadcast,
        blocks: message,
      });
    }
  }
```

### Expected result:
No type error

### Actual result:
```
Argument of type '{ channel: string; thread_ts: string; reply_broadcast: boolean; blocks: KnownBlock[]; }' is not assignable to parameter of type 'ChatPostMessageArguments'.
  Type '{ channel: string; thread_ts: string; reply_broadcast: boolean; blocks: KnownBlock[]; }' is not assignable to type 'TokenOverridable & ChannelAndBlocks & BroadcastedThreadReply & { as_user: true; icon_emoji?: undefined; icon_url?: undefined; } & ... 4 more ... & { ...; }'.
    Type '{ channel: string; thread_ts: string; reply_broadcast: boolean; blocks: KnownBlock[]; }' is not assignable to type 'BroadcastedThreadReply'.
      Types of property 'reply_broadcast' are incompatible.
        Type 'boolean' is not assignable to type 'true'.
```
Note: if I set `reply_broadcast` to `true` or `false` the error goes away.

### Requirements

For general questions/issues about Slack API platform or its server-side, could you submit questions at https://my.slack.com/help/requests/new instead. :bow:

Please read the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

**Comment by filmaj** [comment]
I can reproduce this but I'm not sure how to fix this without loosening up the argument typing.

For reference, [here](https://github.com/slackapi/node-slack-sdk/blob/main/packages/web-api/src/types/request/chat.ts#L75-L92) are the three interfaces that govern this issue: one interface modeling `thread_ts`, and then two others that model the situation for when `reply_broadcast` is `true` vs. `false`.

I was able to reproduce this in our type tests that we have for this repo, particularly those around [this part of the tests](https://github.com/slackapi/node-slack-sdk/blob/main/packages/web-api/test/types/methods/chat.test-d.ts#L281-L286), like this:

```typescript
const bool: boolean;
expectAssignable<Parameters<typeof web.chat.postMessage>>([{
  blocks: [],
  channel: 'C1234',
  thread_ts: '1234.56',
  reply_broadcast: bool, // can send a threaded message and broadcast it, too
}]);
```

... but assigning a value to the `bool` constant then makes the issue go away (changing the above line `const bool: boolean = true;` or `false`). Not sure why this is the case.. not a TypeScript expert but I assumed that modeling both `true` and `false` values for `boolean` would be sufficient to model all possible values of the `boolean` type.

Open to any suggestions on how to address this.

**Comment by filmaj** [comment]
This has been fixed in the just-released web-api v7.3.2
